CHANGELOG.md

```diff
@@ -1,5 +1,22 @@
 # Change Log
 
+## [0.4.0](https://github.com/angelozerr/lsp4xml/milestone/4?closed=1) (March 06, 2019)
+
+### Enhancements
+
+* Modified schema validation messages. See [#181](https://github.com/angelozerr/lsp4xml/issues/181).
+* Preference `xml.format.quotations` to set single vs double quotes for attribute values on format. See [#263](https://github.com/angelozerr/lsp4xml/issues/263).
+* Preference `xml.format.preserveEmptyContent` to preserve a whitespace value in an element's content. See [#273](https://github.com/angelozerr/lsp4xml/issues/273).
+* Removed support for preference `xml.format.joinContentLines`.
+* Compatibility with OSGi and p2. See [#288](https://github.com/angelozerr/lsp4xml/issues/288).
+
+### Bug Fixes
+
+* Fixed memory leak of file handles. See [vscode-xml#298](https://github.com/redhat-developer/vscode-xml/issues/108).
```

Comment:
avoid linking client issues, as we need to keep a neutral stance toward them. Use  https://github.com/angelozerr/lsp4xml/pull/303 instead